### ISSUE TITLE
Unmet optional dependency in ConfigScreenFactory causes a crash

### ISSUE THREAD

**Issue by eggz101** [Prospector/ModMenu#327]
<details>
<summary>crash report </summary>

```
---- Minecraft Crash Report ----
// You should try our sister game, Minceraft!

Time: 02/09/21, 10:38 am
Description: Rendering screen

java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "java.util.Map.get(Object)" is null
	at com.terraformersmc.modmenu.gui.ModsScreen$1.method_25394(ModsScreen.java:151)
	at net.minecraft.class_437.method_25394(class_437.java:107)
	at com.terraformersmc.modmenu.gui.ModsScreen.method_25394(ModsScreen.java:346)
	at net.minecraft.class_757.method_3192(class_757.java:874)
	at net.minecraft.class_310.method_1523(class_310.java:1112)
	at net.minecraft.class_310.method_1514(class_310.java:728)
	at net.minecraft.client.main.Main.main(Main.java:217)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at com.terraformersmc.modmenu.gui.ModsScreen$1.method_25394(ModsScreen.java:151)
	at net.minecraft.class_437.method_25394(class_437.java:107)
	at com.terraformersmc.modmenu.gui.ModsScreen.method_25394(ModsScreen.java:346)

-- Screen render details --
Details:
	Screen name: com.terraformersmc.modmenu.gui.ModsScreen
	Mouse location: Scaled: (121, 303). Absolute: (243.000000, 606.000000)
	Screen size: Scaled: (683, 384). Absolute: (1366, 768). Scale factor of 2.000000

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: Default, Fabric Mods, Obvious_Debris.zip, VanillaTweaks_r594318.zip
Stacktrace:
	at net.minecraft.class_6360.method_36565(class_6360.java:49)
	at net.minecraft.class_310.method_1587(class_310.java:2406)
	at net.minecraft.class_310.method_1514(class_310.java:747)
	at net.minecraft.client.main.Main.main(Main.java:217)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)

-- System Details --
Details:
	Minecraft Version: 1.17.1
	Minecraft Version ID: 1.17.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 16.0.1, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 330434104 bytes (315 MiB) / 1140850688 bytes (1088 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 4
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i5-5300U CPU @ 2.30GHz
	Identifier: Intel64 Family 6 Model 61 Stepping 4
	Microarchitecture: Broadwell (Client)
	Frequency (GHz): 2.30
	Number of physical packages: 1
	Number of physical CPUs: 2
	Number of logical CPUs: 4
	Graphics card #0 name: Intel(R) HD Graphics 5500
	Graphics card #0 vendor: Intel Corporation (0x8086)
	Graphics card #0 VRAM (MB): 1024.00
	Graphics card #0 deviceId: 0x1616
	Graphics card #0 versionInfo: DriverVersion=20.19.15.4963
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 1.60
	Memory slot #0 type: DDR3
	Virtual memory max (MB): 14032.23
	Virtual memory used (MB): 7531.82
	Swap memory total (MB): 6144.00
	Swap memory used (MB): 506.39
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		clear-skies: Clear Skies 1.5.69
		com_velocitypowered_velocity-native: velocity-native 1.1.0-SNAPSHOT
		fabric: Fabric API 0.36.1+1.17
		fabric-api-base: Fabric API Base 0.3.0+a02b446313
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.2.0+2b5c62d013
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.11+c345aea813
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.5+a02b446313
		fabric-command-api-v1: Fabric Command API (v1) 1.1.2+6cefd57713
		fabric-commands-v0: Fabric Commands (v0) 0.2.2+92519afa13
		fabric-containers-v0: Fabric Containers (v0) 0.1.12+a02b446313
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.2+a02b446313
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.5+be9da31013
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.11+6cefd57713
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.1.0+a02b446313
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.9+a722d8c013
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.1+92519afa13
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.7+6cefd57713
		fabric-item-api-v1: Fabric Item API (v1) 1.2.4+a02b446313
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.10+b7ab612113
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.4+a02b446313
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.2+36b77c3e13
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.4.4+a02b44633d
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.4+a02b446313
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.3+92519afa13
		fabric-models-v0: Fabric Models (v0) 0.3.0+a02b446313
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.12+6cefd57713
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.11+a02b446313
		fabric-networking-v0: Fabric Networking (v0) 0.3.2+92519afa13
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.10.9+b7ab612113
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.3+a02b446313
		fabric-particles-v1: Fabric Particles (v1) 0.2.4+a02b446313
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.10+e2961fee13
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.4+5f02c96913
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.8+a02b446313
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.0+a02b446313
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.5+a02b446313
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.13+a02b446313
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.2+92519afa13
		fabric-rendering-v1: Fabric Rendering (v1) 1.6.0+a02b446313
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.7+b7ab61213d
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.4+198a96213d
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.8+a02b446313
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.12+6cefd57713
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.4+a02b446313
		fabric-textures-v0: Fabric Textures (v0) 1.0.6+a02b446313
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.12+b7ab612113
		fabricloader: Fabric Loader 0.11.6
		hydrogen: Hydrogen 0.3
		iris: Iris 1.1.1
		java: OpenJDK 64-Bit Server VM 16
		krypton: Krypton 0.1.4
		litematica: Litematica 0.0.0-dev.20210707.011234
		lithium: Lithium 0.7.3
		logical_zoom: Logical Zoom 0.0.9
		malilib: MaLiLib 0.10.0-dev.23
		minecraft: Minecraft 1.17.1
		minihud: MiniHUD 0.19.0-dev.20210707.150016
		modmenu: Mod Menu 2.0.8
		no_fog: No Fog 1.1.0+1.17.1
		notes: Notes 1.0.7
		sodium: Sodium 0.3.0+IRIS2
		tweakeroo: Tweakeroo 0.10.0-dev.20210710.155746
		xaerominimap: Xaero's Minimap 21.14.1
		xaeroworldmap: Xaero's World Map 1.16.0
	Launched Version: fabric-loader-0.11.6-1.17.1
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: Intel(R) HD Graphics 5500 GL version 3.2.0 - Build 20.19.15.4963, Intel
	Window size: 1366x768
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: 
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Graphics mode: fast
	Resource Packs: vanilla, Fabric Mods, file/Obvious_Debris.zip (incompatible), file/VanillaTweaks_r594318.zip
	Current Language: English (US)
	CPU: 4x Intel(R) Core(TM) i5-5300U CPU @ 2.30GHz
```

</details>

**Comment by jackassmc** [comment]
This is caused by the Notes mod not declaring Cloth Config API as a dependency and launching without it. You can download it from [Curseforge](https://www.curseforge.com/minecraft/mc-mods/cloth-config) to fix the crash.

A mod having an optional config screen that depends on an optional library is a valid use case and crashing because of it is a bug in Mod Menu. Thank you for reporting this bug! A new update with a fix for this issue will be available soon.

**Comment by jackassmc** [comment]
Fixed in 0cc12ae09bc62262389b4639e623c5ee95481111/[v1.16.16](https://github.com/TerraformersMC/ModMenu/releases/tag/v1.16.16) and a4ecedbf357c902e16db95cea542d000f9a832e4/[v2.0.9](https://github.com/TerraformersMC/ModMenu/releases/tag/v2.0.9) .

**Comment by eggz101** [comment]
thank you
